### PR TITLE
[MNOE-582] Remove impersonation request if user modify account setting to authorise admin to access its account

### PR DESCRIPTION
**Pull request by x4d3**

### CODE REVIEW

File: api/lib/mno_enterprise/concerns/controllers/jpi/v1/user_access_requests_controller.rb [hunk 1]
```diff
@@ -23,6 +23,18 @@ def last_access_request
 
   # POST /mnoe/jpi/v1/user_access_requests
   def create
+    access_duration = params.require(:access_duration)
+    user = current_user.load_required(:user_access_requests, :'user_access_requests.requester')
+    user.user_access_requests.each do |request|
```
Comment (BrunoChauvet) [inline api/lib/mno_enterprise/concerns/controllers/jpi/v1/user_access_requests_controller.rb]: I would move this logic inside the class `UserAccessRequest` as a class method and then call something like
```ruby
MnoEnterprise::UserAccessRequest.approve_pending_requests(user)
```

File: api/lib/mno_enterprise/concerns/mailers/system_notification_mailer.rb [hunk 2]
```diff
@@ -211,6 +211,21 @@ def access_approved(user_access_request_id)
     )
   end
 
+  def access_approved_all(user_id, requester_id, access_duration)
```
Comment (BrunoChauvet) [inline api/lib/mno_enterprise/concerns/mailers/system_notification_mailer.rb]: Is it possible to pass the objects `user` and `requester` as they have already been loaded before?

File: api/spec/controllers/mno_enterprise/jpi/v1/user_access_requests_controller_spec.rb [hunk 3]
```diff
@@ -33,6 +33,19 @@ module MnoEnterprise
...
+      let!(:stubs) {
+        [
+          stub_api_v2(:post, "/user_access_requests", user_access_request),
+          stub_api_v2(:get, "/users/#{user.id}", user, [:user_access_requests, :'user_access_requests.requester']),
+          stub_api_v2(:get, "/user_access_requests/#{user_access_request.id}", user_access_request, [:requester])
+        ]
+      }
+      subject { post :create, access_duration: 'UNTIL_REVOKED'}
+      it_behaves_like 'jpi v1 protected action'
+    end
```
Comment (BrunoChauvet) [inline api/spec/controllers/mno_enterprise/jpi/v1/user_access_requests_controller_spec.rb]: Can you add a scenario where there are pending `user_access_requests`?

File: api/lib/mno_enterprise/concerns/controllers/jpi/v1/user_access_requests_controller.rb (hunk 1 above)
Comment (x4d3) [reply api/lib/mno_enterprise/concerns/controllers/jpi/v1/user_access_requests_controller.rb]: Ok

File: api/lib/mno_enterprise/concerns/mailers/system_notification_mailer.rb (hunk 2 above)
Comment (x4d3) [reply api/lib/mno_enterprise/concerns/mailers/system_notification_mailer.rb]: I don't trust serialization on BaseResource Object.

**Comment by x4d3** [comment]
@BrunoChauvet I've applied your comments.

**Pull request merged**